src/Npgsql/NpgsqlDataAdapter.cs

```diff
@@ -137,6 +140,74 @@ protected override void OnRowUpdating(RowUpdatingEventArgs value)
             get => (NpgsqlCommand?)base.InsertCommand;
             set => base.InsertCommand = value;
         }
+
+        internal Task<int> Fill(DataTable dataTable, bool async, CancellationToken cancellationToken = default)
+        {
+            if (async)
```

Comment:
To be honest, if we're introducing our own implementation of FillAsync, I'd just have the sync code go through that, with the standard async flag. This way we don't have two different implementations for sync/async which may behave differently.